hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -691,11 +703,19 @@ onOpenNetworkReqSn env ledger st otherParty sn requestedTxs =
       then continue
       else Wait $ WaitOnSnapshotNumber seenSn
 
+  waitSeenTxs continue =
+    if fromList requestedTxIds `Set.isSubsetOf` fromList (txId <$> seenTxs)
+      then continue
+      else Wait $ WaitOnSeenTxs $ toList $ fromList requestedTxIds `Set.difference` fromList (txId <$> seenTxs)
```

Comment:
Could: change the type to a `Set (TxIdType tx)` and not convert `toList` here.

Could: use `\\`